File: go/1414.go
func findMinFibonacciNumbers(k int) int {
    fib := []int{1, 1}
    for fib[len(fib) - 1] < k {
        fib = append(fib, fib[len(fib) - 1] + fib[len(fib) - 2])
    }
    ans := 0
    for i := len(fib) - 1; i > 0; i-- {
        if k >= fib[i] {
            k -= fib[i]
            ans += 1
        }
    }
    return ans
}

File: rust/1414.rs
impl Solution {
    pub fn find_min_fibonacci_numbers(k: i32) -> i32 {
        let mut k = k;
        let mut fib = vec![1, 1];
        loop {
            let x = fib[fib.len() - 1] + fib[fib.len() - 2];
            if x > k {
                break;
            }
            fib.push(x);
        }
        let mut ans = 0;
        for x in fib.iter().rev() {
            if k >= *x {
                k -= x;
                ans += 1;
            }
            if k == 0 {
                break;
            }
        }
        ans
    }
}
